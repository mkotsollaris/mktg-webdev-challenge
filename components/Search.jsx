const Search = () => {
    return <>
        <form className="example" action="action_page.php">
            <input type="text" placeholder="Search people by name" name="search" />
        </form>
    </>
}

export default Search;