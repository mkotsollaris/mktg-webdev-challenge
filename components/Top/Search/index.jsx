import { useContext } from 'react'
import style from './style.module.css'
import searchIconSrc from '../../../public/img/search.svg'
import AppContext from '../../../context/AppContext'

const Search = () => {
  const { setSearchValue } = useContext(AppContext)

  return (
    <>
      <div className={style['search-grid']}>
        <div className={style['search-icon']}>
          <img src={searchIconSrc} alt="search" />
        </div>
        <div>
          <input
            onChange={(e) => setSearchValue(e.target.value.toLowerCase())}
            type="text"
            placeholder="Search people by name"
          />
        </div>
      </div>
    </>
  )
}

export default Search
