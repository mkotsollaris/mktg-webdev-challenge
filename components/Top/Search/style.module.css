.search-grid {
  display: grid;
  border: 1px solid gray;
  grid-template-columns: 2rem 20rem;
  border-radius: 2rem;
  justify-content: center;
  align-content: center;
  align-items: center;
  padding: 0.5rem;
  & :focus {
    outline: none;
  }
  &:nth-child(3) {
    margin-bottom: 1rem;
  }
  @media (max-width: 400px) {
    composes: search-grid;
    grid-template-columns: 10% 90%;
  }
}

.search-icon {
  display: grid;
  justify-content: center;
  align-content: center;
  align-items: center;
}

.search-grid input {
  border: none;
  width: 100%;
  height: 2rem;
}
