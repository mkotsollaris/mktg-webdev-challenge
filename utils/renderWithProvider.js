import { render } from '@testing-library/react'
import AppContext from '../context/AppContext'

function renderWithProvider(props, Component) {
  return render(
    <AppContext.Provider value={props}>
      <Component />
    </AppContext.Provider>
  )
}

export default renderWithProvider
