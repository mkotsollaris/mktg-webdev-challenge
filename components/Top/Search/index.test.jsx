import Search from '.'
import renderWithProvider from '../../../utils/renderWithProvider'

it('renders', () => {
  const { asFragment } = renderWithProvider({}, Search)
  expect(asFragment()).toMatchSnapshot()
})
