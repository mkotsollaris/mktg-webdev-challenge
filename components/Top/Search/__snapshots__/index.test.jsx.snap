// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    role="search"
  >
    <div>
      <img
        alt="search"
        src="[object Object]"
      />
    </div>
    <div>
      <input
        placeholder="Search people by name"
        type="search"
      />
    </div>
  </div>
</DocumentFragment>
`;
